src/open_producten/producttypen/views.py

```
@@ -41,9 +43,36 @@
 from open_producten.producttypen.serializers.producttype import (
     ProductTypeTranslationSerializer,
 )
+from open_producten.utils.filters import FilterSet
 from open_producten.utils.views import OrderedModelViewSet, TranslatableViewSetMixin
 
 
+class ProductTypeFilterSet(FilterSet):
+    regex = r"^\[([^:\[\]]+):([^:\[\]]+)\]$"  # [key:value] where the key and value cannot contain `[`, `]` or `:`
+
+    externe_code = django_filters.CharFilter(
+        method="filter_by_externe_code", validators=[RegexValidator(regex)]
+    )
+
+    uniforme_product_naam = django_filters.CharFilter(
+        field_name="uniforme_product_naam__naam"
+    )
+
+    def filter_by_externe_code(self, queryset, name, value):
+        value_list = value.strip("[]").split(":")
+
+        if len(value_list) != 2:
+            return queryset
```

I think it's better to raise a 400 error if the input is correct, otherwise the impression is raised that the returned result was properly filtered